### PR TITLE
ISSUE-41 - overlays are sometimes not positioned correctly

### LINKED ISSUE #41: Overlays are sometimes not positioned correctly
see http://www.boots.com/

similar problem with body position: relative and margins

### PR DESCRIPTION
**Pull request by tabanliviu**
fixes #41 
update on scroll position changes
use correct calculation for top/left on overlays

### CODE REVIEW

File: test/public.spec.js [hunk 1]
```diff
@@ -144,6 +144,38 @@ describe('imageChecker', () => {
...
+        let intervalFn = window.setInterval.calls.mostRecent().args[0];
+        createDomNodes([
+          createImg()
+        ]);
+        window.scrollX += 1;
+
+        intervalFn();
+
+        let imageOverlays = document.querySelectorAll('.ncc-image-checker-overlay');
+        expect(imageOverlays.length).toEqual(2);
```
Comment (MalcolmSmithUK) [inline test/public.spec.js]: @tabanliviu COVERAGE - It's not clear why the assertion is related to the describe block.

File: test/public.spec.js [hunk 2]
```diff
@@ -467,7 +499,7 @@ describe('imageChecker', () => {
     if (options.queryParams) {
       bigImg.src += `?${options.queryParams}`;
     }
-    bigImg.style = 'display: block;width: 200px;height: 160px;';
+    bigImg.style = 'display: block;width: 300px;height: 240px;';
```
Comment (MalcolmSmithUK) [inline test/public.spec.js]: @tabanliviu COVERAGE - This indicates a missing test case.

**Review by MalcolmSmithUK — CHANGES_REQUESTED** [review]

**Comment by tabanliviu** [comment]
@MalcolmSmithUK updated with requested changes

**Comment by Ian-Hi** [comment]
http://www.msn.com/en-gb/news/uknews/woman-steals-her-stolen-bike-back-from-thief-who-advertised-it-on-facebook/ar-BBEvNtA?li=BBoPRmx is vertically mis-aligned above 1262, but fine below that width

**Review by MalcolmSmithUK — DISMISSED** [review]

**Comment by tabanliviu** [comment]
@Ian-Hi Updated branch can you please have a look? I've also added a manual test page that replicates the issue. see `test/manual` folder

**Review by MalcolmSmithUK — DISMISSED** [review]

File: app/public.js [hunk 3]
```diff
@@ -269,19 +287,29 @@
...
-    if (elem.x && elem.y) {
-      location.top = elem.y;
-      location.left = elem.x;
-    } else if (elem.offsetParent) {
-      do {
-        location.top += elem.offsetTop;
-        location.left += elem.offsetLeft;
-        elem = elem.offsetParent;
-      } while (elem);
+    if (elem.getBoundingClientRect) {
```
Comment (MalcolmSmithUK) [inline app/public.js]: @tabanliviu MAINTAINABILITY - No need for the guard here - plus potential for consolidating `getImagesInfoParent` and `getElementTopLeft`.

**Review by MalcolmSmithUK — CHANGES_REQUESTED** [review]

**Review by MalcolmSmithUK — APPROVED** [review]

**Pull request merged**